core/blockchain.go

```diff
@@ -2336,3 +2337,45 @@ func (bc *BlockChain) SubscribeLogsEvent(ch chan<- []*types.Log) event.Subscript
 func (bc *BlockChain) SubscribeBlockProcessingEvent(ch chan<- bool) event.Subscription {
 	return bc.scope.Track(bc.blockProcFeed.Subscribe(ch))
 }
+
+func (bc *BlockChain) RecoverRandomnessCache(commitment common.Hash, commitmentBlockHash common.Hash) error {
+	if istEngine, isIstanbul := bc.engine.(consensus.Istanbul); isIstanbul {
+
+		blockHashIter := commitmentBlockHash
```

Comment:
i would add a log message here, it's quite a rare thing, and can take a decent amount of time, so an `INFO` msg would be nice, to inform the user what's happening